src/autocomplete.js

```diff
@@ -50,6 +58,13 @@ class Autocomplete extends Component {
   input = null
   items = []
 
+  getState() {
+    return Object.keys(this.state).reduce((state, key) => {
+      state[key] = this.props[key] == null ? this.state[key] : this.props[key]
+      return state
+    }, {})
+  }
```

Comment:
Great suggestion, really simplifies things. Though I would prefer to not spread all the props on that object if I can avoid it. The only use case I care about is the state for this function.

But you do make a great point here:

> This nicely overwrites with props if defined (the details are a little different to this method, because if a prop is available and set to null it would still overwrite the state, but I believe that's what a user would expect)

So I'm going to change that to use `this.props[key] === undefined` so the user can set it to null if they wish.